numba/parfors/parfor_lowering.py

```diff
@@ -738,13 +738,25 @@ def find_setitems_body(setitems, itemsset, loop_body, typemap):
     for label, block in loop_body.items():
         find_setitems_block(setitems, itemsset, block, typemap)
 
+def empty_container_allocator_hoist(inst, dep_on_param, call_table, hoisted,
+                                    not_hoisted, typemap, stored_arrays):
+    if (isinstance(inst, ir.Assign) and
+        isinstance(inst.value, ir.Expr) and
+        inst.value.op == 'call' and
+        inst.value.func.name in call_table):
+        call_list = call_table[inst.value.func.name]
+        if call_list == ['empty', np]:
+            return _hoist_internal(inst, dep_on_param, call_table, hoisted,
+                                   not_hoisted, typemap, stored_arrays)
+    return False
```

Comment:
Could this use this: https://github.com/numba/numba/blob/master/numba/core/ir_utils.py#L703-L726 ?

Reply:
No, I don't think it would be correct to use "is_pure" here because this here is specifically about the allocation of empty containers.

Reply:
ok, I see, thanks, whilst the code is very similar the purpose is different.